src/include/sof/lib/cps-budget-manager.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Krzysztof Frydryk <frydryk.krzysztof@intel.com>
+ */
+
+#ifndef __SOF_LIB_CPS_BUDGET_MANAGER_H__
+#define __SOF_LIB_CPS_BUDGET_MANAGER_H__
+
+#include <rtos/spinlock.h>
+#include <sof/sof.h>
+#include <stdint.h>
+
+struct cps_budget_data {
+	int kcps_consumtion[CONFIG_CORE_COUNT];
```

Comment:
This is not recommended on multicore as the array[CORE_COUNT] can easily be clobbered. This structure should either use coherent API or be uncache only (with the correct SPARSE types).